transcom/mymove
# B 20034 int

## [20034](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A950001)

[Review #1](https://github.com/transcom/mymove/pull/12595)
[Review #2](https://github.com/transcom/mymove/pull/12636)

## Summary

This PR is to fix some pre-test issues in the main review, that passed during the integration review(s). This review also includes a new UPDATE statement in the migrations. This is to automatically update the application_parameters table from supporting just validation_codes to supporting the new param_name and param_value structure. The INSERT for the validation_code values was initially done manually in testing and production environments.

### How to test

Couple of setup things you need to do first:
- `make db_dev_migrate` (this will create the `application_parameters` table
- You'll need to fill in a row with a value in `parameter_name`, 'parameter_value' and `id`, `updated_at` and `created_at` will auto populate
- In your `.envrc`, set `export FEATURE_FLAG_VALIDATION_CODE_REQUIRED=true`
- Now fire up your `client` and `server`


To test db update of validation_code to param_name and param_value (this is meant to automatically update real environments to support the new table structure):
1. add a new entry to your local DB application_parameters table with an ID column value of '30457d8e-b108-4173-a757-4acc17983904', and an entry into VALIDATION_CODE column  with whatever value you want to use. I used test1234. 

Sample SQL statement for insert:
INSERT INTO application_parameters(id, validation_code)
VALUES ('30457d8e-b108-4173-a757-4acc17983904', 'test1234');

3. Run : update application_parameters set parameter_value = validation_code, parameter_name = 'validation_code' where id = '30457d8e-b108-4173-a757-4acc17983904' 
4. View the table to verify that the new columns param_name and param_value have been populated with param_name='validation_code' and param_value equal to the value you chose.


UI tests:
**For the happiest path**
1. Access MM as a customer
5. Login as a brand new customer
6. You should now see the validation code screen
7. Input the code you slapped in the `application_parameters` table
8. It should now let you progress through the profile set up

**For the saddest path**
1. Access MM as a customer
3. Login as a brand new customer
4. You should now see the validation code screen
5. Input anything other than the code you put in the database
6. You should see an error screen and not be able to progress
9. You can have as many attempts as you want when inputting the code (this was a requirement from Kipp)

## Screenshots
![Screenshot 2024-04-12 at 11 07 01 AM](https://github.com/transcom/mymove/assets/136510600/e8e522af-ecba-490e-b1d3-f3de93b81b2d)

Mobile
![Screenshot 2024-04-12 at 11 21 56 AM](https://github.com/transcom/mymove/assets/136510600/c7adffae-d824-4311-ae8d-bdcb430d3074)

![Screenshot 2024-04-12 at 11 12 31 AM](https://github.com/transcom/mymove/assets/136510600/c72a663b-caf1-4fa2-848e-432da5809dd3)

Mobile
![Screenshot 2024-04-12 at 11 21 38 AM](https://github.com/transcom/mymove/assets/136510600/bad88699-4e8f-4f20-a0fb-e8f4532d5639)
